FYI, lots of breaks, e.g.
```
error CS8767: (NETCORE_ENGINEERING_TELEMETRY=Build) Nullability of reference types in type of parameter 'buffer' of 'long DbDataRecord.GetBytes(int i, long dataIndex, byte[] buffer, int bufferIndex, int length)' doesn't match implicitly implemented member 'long IDataRecord.GetBytes(int i, long fieldOffset, byte[]? buffer, int bufferoffset, int length)' (possibly because of nullability attributes).
```

--- a/src/libraries/System.Data.Common/ref/System.Data.Common.cs
+++ b/src/libraries/System.Data.Common/ref/System.Data.Common.cs
@@ -2202,9 +2202,9 @@ namespace System.Data.Common
         public abstract object this[string name] { get; }
         public abstract bool GetBoolean(int i);
         public abstract byte GetByte(int i);
-        public abstract long GetBytes(int i, long dataIndex, byte[] buffer, int bufferIndex, int length);
+        public abstract long GetBytes(int i, long dataIndex, byte[]? buffer, int bufferIndex, int length);
         public abstract char GetChar(int i);
-        public abstract long GetChars(int i, long dataIndex, char[] buffer, int bufferIndex, int length);
+        public abstract long GetChars(int i, long dataIndex, char[]? buffer, int bufferIndex, int length);
         public System.Data.IDataReader GetData(int i) { throw null; }
         public abstract string GetDataTypeName(int i);
         public abstract System.DateTime GetDateTime(int i);

--- a/src/libraries/System.Data.Common/src/System/Data/Common/DataRecordInternal.cs
+++ b/src/libraries/System.Data.Common/src/System/Data/Common/DataRecordInternal.cs
@@ -100,7 +100,7 @@ namespace System.Data.Common
             return ((byte)_values[i]);
         }
 
-        public override long GetBytes(int i, long dataIndex, byte[] buffer, int bufferIndex, int length)
+        public override long GetBytes(int i, long dataIndex, byte[]? buffer, int bufferIndex, int length)
         {
             int cbytes = 0;
             int ndataIndex;
@@ -170,7 +170,7 @@ namespace System.Data.Common
 
         public override char GetChar(int i) => ((string)_values[i])[0];
 
-        public override long GetChars(int i, long dataIndex, char[] buffer, int bufferIndex, int length)
+        public override long GetChars(int i, long dataIndex, char[]? buffer, int bufferIndex, int length)
         {
             // if the object doesn't contain a char[] then the user will get an exception
             string s = (string)_values[i];

--- a/src/libraries/System.Data.Common/src/System/Data/Common/DbDataRecord.cs
+++ b/src/libraries/System.Data.Common/src/System/Data/Common/DbDataRecord.cs
@@ -19,11 +19,11 @@ namespace System.Data.Common
 
         public abstract byte GetByte(int i);
 
-        public abstract long GetBytes(int i, long dataIndex, byte[] buffer, int bufferIndex, int length);
+        public abstract long GetBytes(int i, long dataIndex, byte[]? buffer, int bufferIndex, int length);
 
         public abstract char GetChar(int i);
 
-        public abstract long GetChars(int i, long dataIndex, char[] buffer, int bufferIndex, int length);
+        public abstract long GetChars(int i, long dataIndex, char[]? buffer, int bufferIndex, int length);
 
         public IDataReader GetData(int i) => GetDbDataReader(i);
 

